src/components/SelectQuiz.tsx

```diff
@@ -3,25 +3,33 @@ import React from "react";
 interface SelectQuizProps {
   startQuiz: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
   selectQuizArr: number[];
+  selectedCategory: string; // Add the selectedCategory prop
+  totalQuestions: number; // Add the totalQuestions prop
 }
 
 const SelectQuiz: React.FC<SelectQuizProps> = SelectQuizProps => {
+  
+  const availableQuizLengths = SelectQuizProps.selectQuizArr.filter((length) => length <= SelectQuizProps.totalQuestions);
+
   return (
     <div className="select-quiz-styles">
       <h2 className="quiz-heading">Choose a length for the Quiz</h2>
       <div className="w-25 select-btn-div">
-        {SelectQuizProps.selectQuizArr.map(
-          (choice: number, index: number, array: number[]) => (
-            <button
-              className="select-btns"
-              onClick={e => SelectQuizProps.startQuiz(e)}
-              value={choice}
-              key={index}
-            >
-              {index == array.length - 1 ? `All (${choice})` : `${choice}`}
+      {availableQuizLengths.map((choice: number, index: number) => (
+  <button
+    className="select-btns"
+    onClick={(e) => SelectQuizProps.startQuiz(e)}
+    value={choice}
+    key={index}
+  >
+    {choice}
+  </button>
+))}
+           
+            <button className="select-btns" onClick={SelectQuizProps.startQuiz} value={SelectQuizProps.totalQuestions}>
+              All ({SelectQuizProps.totalQuestions})
             </button>
```

Comment:
It looks like the formatting is a little off here
I would suggest running `npm run format`